### ISSUE TITLE
add_membership option for gen_udp:open IPv6 doesn't work

### ISSUE THREAD

**Issue by mats9** [erlang/otp#8176]
**Describe the bug**
Option  add_membership for function gen_udp:open doesn't work for IPv6. The multicast address doesn't get added to the multicast table, and no traffic is received.

**To Reproduce**
```
Address = {65282,0,0,0,0,0,1,2}, % ff02::1:2
Index = 2,
Addr = #{family => inet6, port => 547, addr => Address, scope_id => Index},
AddMemberOpt = {add_membership, {Address, Index}},
{ok, Socket} = gen_udp:open(0, [binary, {ifaddr, Addr}, AddMemberOpt]),
```
Gives result:
```
> ip -6 maddr show dev enp1s0
2:      enp1s0
        inet6 ff02::1:3
        inet6 ff02::fb
        inet6 ff02::1:ff00:193
        inet6 ff02::1:ff2c:867c
        inet6 ff02::1
        inet6 ff01::1
```
ff02::1:2 is missing.
**Expected behavior**
Using raw format instead it works:
```
Address = {65282,0,0,0,0,0,1,2}, % ff02::1:2
Index = 2,
MaBin = <<65282:16,0:16,0:16,0:16,0:16,0:16,1:16,2:16>>,
IndexBin = <<Index:32/native>>,
IPPROTO_IPV6 = 41,
IPV6_ADD_MEMBERSHIP = 20,
Addr = #{family => inet6, port => 547, addr => Address, scope_id => Index},
AddMemberOpt = {raw, IPPROTO_IPV6, IPV6_ADD_MEMBERSHIP, <<MaBin/binary, IndexBin/binary>>},
{ok, Socket} = gen_udp:open(0, [binary, {ifaddr, Addr}, AddMemberOpt]),
```
Gives result:
```
> ip -6 maddr show dev enp1s0
2:      enp1s0
        inet6 ff02::1:2
        inet6 ff02::1:3
        inet6 ff02::fb
        inet6 ff02::1:ff00:193
        inet6 ff02::1:ff2c:867c
        inet6 ff02::1
        inet6 ff01::1
```
Now ff02::1:2 is in table, and traffic is received.
**Affected versions**
OTP-26.2.2

**Comment by bmk** [comment]
https://github.com/bmk/otp/pull/new/bmk/kernel/20240312/add_membership_ipv6

**Comment by bmk** [comment]
Fix released with OTP 26.2.4
